drivers/kw2xrf/kw2xrf.c

```diff
@@ -0,0 +1,869 @@
+/*
+ * Copyright (C) 2015 PHYTEC Messtechnik GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     kw2xrf
+ * @{
+ * @file        kw2xrf.c
+ * @brief       Basic functionality of kw2xrf driver
+ *
+ * @author      Johann Fischer <j.fischer@phytec.de>
+ * @author      Jonas Remmert <j.remmert@phytec.de>
+ * @}
+ */
+#include "crash.h"
+#include "kw2xrf.h"
+#include "kw2xrf_spi.h"
+#include "kw2xrf_reg.h"
+#include "kw2xrf_internal.h"
+#include "hwtimer.h"
+#include "kernel_types.h"
+#include "netdev/802154.h"
+#include "netdev/base.h"
+#include "thread.h"
+#include "periph_conf.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+static uint16_t radio_pan;
+static uint8_t  radio_channel;
+static uint16_t radio_address;
+static uint64_t radio_address_long;
+
+#define MKW2XDRF_OUTPUT_POWER_MAX  8
+#define MKW2XDRF_OUTPUT_POWER_MIN  (-35)
+
+#define MKW2XDRF_PA_RANGE_MAX      31
+#define MKW2XDRF_PA_RANGE_MIN      3
+
+/* event type to signalize receive event */
+/* TODO: Event_Nr used from cc2420 driver */
+#define MKW2XDRF_NETDEV_EVENT_RX  (3452342219)
+
+/* default source address length for sending in number of byte */
+static size_t _default_src_addr_len = 2;
+
+/* PLL integer lookup table */
+static const uint8_t pll_int_lt[16] = {
```

Comment:
I would let the compiler put in the size of the array... Its just a source for error later on...